eras/shelley/impl/src/Cardano/Ledger/Shelley/LedgerState/IncrementalStake.hs

```diff
@@ -110,20 +123,28 @@ incrementalAggregateUtxoCoinByCredential ::
   forall era.
   EraTxOut era =>
   PParams era ->
-  (Coin -> Coin) ->
+  Mode ->
   UTxO era ->
   IncrementalStake (EraCrypto era) ->
   IncrementalStake (EraCrypto era)
 incrementalAggregateUtxoCoinByCredential pp mode (UTxO u) initial =
   Map.foldl' accum initial u
   where
     keepOrDelete new Nothing =
-      case mode new of
+      case new of
         Coin 0 -> Nothing
         final -> Just final
     keepOrDelete new (Just old) =
-      case mode new <> old of
+      case addCoin mode new old of
+        Coin 0 -> Nothing
+        final -> Just final
+    keepOrDeleteCompact new Nothing =
+      case new of
         Coin 0 -> Nothing
+        final -> Just (compactCoinOrError final)
```

Comment:
It will actually be more efficient to call `compactCoinTxOutL` instead of `coinTxOutL` below, this will avoid the partial conversion here. However this will require that `ptrMap` also holds compact coins as I mentioned in another comment
```suggestion
        final -> Just final
```

Reply:
So using compactCoinTxOutL , and changing ptrMap to use (CompactForm Coin) is brilliant.  There is no mention of coin at all.